anyblok_marshmallow/fields.py

```diff
@@ -297,6 +297,71 @@ def _validate(self, value):
                         (self.key, value, Model))
 
 
+class UniqueField(Field):
+
+    """Adds schema validation for ensuring uncity of a considered field on the
+       considered model.
+
+    :param str model: The database model to query
+    :param key: The database column to query for `value`.
+    """
+
+    def __init__(self, cls_or_instance_type=Str, model=None,
+                 key=None, *args, **kwargs):
+        self.model = model
+        self.key = key
+
+        if isinstance(cls_or_instance_type, type):
+            if not issubclass(cls_or_instance_type, FieldABC):
+                raise ValueError('The type of the list elements '
+                                 'must be a subclass of '
+                                 'marshmallow.base.FieldABC')
+            self.container = cls_or_instance_type()
+        else:
+            if not isinstance(cls_or_instance_type, FieldABC):
+                raise ValueError('The instances of the list '
+                                 'elements must be of type '
+                                 'marshmallow.base.FieldABC')
+            self.container = cls_or_instance_type
+
+        super(UniqueField, self).__init__(*args, **kwargs)
+
+    def _add_to_schema(self, field_name, schema):
+        super(UniqueField, self)._add_to_schema(field_name, schema)
+        self.container.parent = self
+        self.container.name = field_name
+
+    def _serialize(self, value, attr, obj):
+        return self.container._serialize(value, attr, obj)
+
+    def _deserialize(self, value, attr, data, **kwargs):
+        return self.container.deserialize(value, attr=attr, data=data)
+
+    def _validate(self, value):
+        registry = self.context['registry']
+        Model = registry.get(self.model)
+
+        if not value:
+            raise ValidationError("Field may not be null.")
+
+        if isinstance(self.container, List):
```

Comment:
This structure is bizarre, because the notion of fields.List means that the field is multiple whereas it is the value.

it is as if to test the color of the steering wheel of a car, it is considered that a car has several wheels.

personally I am against

Reply:
Ok right I guess I got it. But then since I adapted my code regarding the InstanceField field, why is it different, since this field implements some checks for fields.List cases ?

Reply:
For me, it is not a list of entry, and it must be only  for the post. I think it have not be a field but a validator and only for the serialization.

I will see this evening, If I find a better way